okapi-core/src/main/java/org/folio/okapi/util/DepResolution.java

```diff
@@ -128,63 +128,59 @@ public static void getLatestProducts(int limit, List<ModuleDescriptor> mdl) {
 
   /**
    * Check if md's required and optional interfaces are provided by set of modules.
-   * The interface version is not checked, only the interface name.
-   * Only interfaces listed in allProvided are checked, this allows to exclude some optional
-   * interfaces from the check.
+   * Only interfaces listed in allProvided are checked, this allows for the exclusion of
+   * some optional interfaces from the check.
    *
-   * @param modules     the modules that check against
-   * @param allProvided interfaces for all modules
+   * @param modules     the modules to check against
    * @param md          module to check.
    * @return            true if interface requirements are met
    */
-  static boolean moduleDepProvided(
-      List<ModuleDescriptor> modules, Set<String> allProvided,
-      ModuleDescriptor md) {
+  public static boolean moduleDepProvided(
+      Collection<ModuleDescriptor> modules, ModuleDescriptor md) {
 
-    for (InterfaceDescriptor req : md.getRequiresOptionalList()) {
-      if (allProvided.contains(req.getId())) {
-        boolean found = false;
-        for (ModuleDescriptor md1 : modules) {
-          InterfaceDescriptor[] providesList = md1.getProvidesList();
-          for (InterfaceDescriptor prov : providesList) {
-            if (prov.isRegularHandler() && prov.getId().equals(req.getId())) {
-              found = true;
-            }
+    for (InterfaceDescriptor req : md.getRequiresList()) {
+      boolean found = false;
+      for (ModuleDescriptor md1 : modules) {
+        InterfaceDescriptor[] providesList = md1.getProvidesList();
+        for (InterfaceDescriptor prov : providesList) {
+          if (prov.isRegularHandler() && prov.isCompatible(req)) {
+            found = true;
           }
         }
-        if (!found) {
-          return false;
-        }
+      }
+      if (!found) {
+        return false;
       }
     }
     return true;
   }
 
-  static void topoSort(List<ModuleDescriptor> modules) {
-    List<ModuleDescriptor> result = new LinkedList<>();
+  /**
+   * Check if md's provided interfaces are required by any of the modules.
```

Comment:
Check if at least one of md's provided interfaces is a required or optional dependency of any of the modules.

Reply:
"of any of the" triggers a warning here :-) I understand it.